tests/Integration/Schema/Directives/WithAvgDirectiveTest.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+namespace Tests\Integration\Schema\Directives;
+
+use function factory;
+use Illuminate\Support\Facades\DB;
+use Tests\DBTestCase;
+use Tests\Utils\Models\Author;
+use Tests\Utils\Models\Book;
+
+class WithAvgDirectiveTest extends DBTestCase
+{
+    public function testEagerLoadsRelationAvg(): void
+    {
+        $this->schema = /** @lang GraphQL */ '
+        type Query {
+            authors: [Author!] @all
+        }
+
+        type Book {
+            title: String!
+            price: Int!
+        }
+
+        type Author {
+            name: String!
+            books:[Book!]! @belongsToMany
+            books_avg_price: Int!
```

Comment:
*Sigh*

As I have now repeated multiple times: this usage of the `@withAvg` directive is wrong! If you need to get the actual aggregate value directly as the field result, a simpler version of this directive should be used. It would be called `@avg`.

The `@with` directives are meant to load a relation (or a relation aggregate) as a side effect when a certain field is requested, assuming the field resolver subsequently makes use of that field.

Even if this test is technically working, it is setting a wrong example. What you are doing here would be much simpler with `@avg`, and the test does not demonstrate proper use of `@withAvg`. Consider the test case for `@withCount` and how the assertion for proper loading is made there.